addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -778,6 +778,74 @@ const actions = {
         }
         thread.message_unread_counter = 0;
     },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationAccept({ dispatch, state }, messageLocalIds) {
+        return dispatch('_moderateMessages', messageLocalIds, 'accept');
+    },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationAllow({ dispatch, state }, messageLocalIds) {
+        return dispatch('_moderateMessages', messageLocalIds, 'allow');
+    },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationBan({ dispatch, state }, messageLocalIds) {
+        return dispatch('_moderateMessages', messageLocalIds, 'ban');
+    },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationDiscard({ dispatch, state }, messageLocalIds) {
+        return dispatch('_moderateMessages', messageLocalIds, 'discard');
+    },
+    /**
+     * @param {Object} param0
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    moderationMessageCheck({ state }, messageLocalIds) {
+        for (const messageLocalId of messageLocalIds) {
+            state.messages[messageLocalId].isModerationChecked = true;
```

Comment:
this looks wrong:
- check state of messages should not be moderation specific (even though it's used just for moderation at the moment)
- this is `thread` and `domain` specific

Reply:
done